tests/query/processors/test_uniq_in_select_and_having.py

```diff
@@ -0,0 +1,97 @@
+from copy import deepcopy
+
+import pytest
+
+from snuba.clickhouse.query import Query as ClickhouseQuery
+from snuba.query.expressions import Column, FunctionCall, Literal
+from snuba.query.processors.uniq_in_select_and_having import (
+    MismatchedAggregationException,
+    UniqInSelectAndHavingProcessor,
+)
+from snuba.request.request_settings import HTTPRequestSettings
+from snuba.state import set_config
+from tests.query.processors.query_builders import build_query
+
+
+def uniq_expression(alias: str = None, column_name: str = "user") -> FunctionCall:
+    return FunctionCall(
+        None,
+        "greater",
+        (
+            FunctionCall(alias, "uniq", (Column(None, None, column_name),)),
+            Literal(None, 1),
+        ),
+    )
+
+
+INVALID_QUERY_CASES = [
+    pytest.param(
+        build_query(
+            selected_columns=[
+                Column("_snuba_project_id", None, "project_id"),
+                Column(None, None, "transaction_name"),
+            ],
+            condition=None,
+            having=uniq_expression(),
+        ),
+        id="uniq in having, not in select",
+    ),
+    pytest.param(
+        build_query(
+            selected_columns=[
+                Column("_snuba_project_id", None, "project_id"),
+                Column(None, None, "transaction_name"),
+                uniq_expression(alias="my_alias", column_name="some_column"),
+            ],
+            condition=None,
+            having=uniq_expression(alias="my_alias", column_name="some_other_column"),
+        ),
+        id="same aggregation, different columns",
+    ),
+]
+
+VALID_QUERY_CASES = [
+    pytest.param(
+        build_query(
+            selected_columns=[
+                Column("_snuba_project_id", None, "project_id"),
+                Column(None, None, "transaction_name"),
+                # the parser guarantees alias expansion
+                uniq_expression(alias="my_alias"),
+            ],
+            condition=None,
+            having=uniq_expression(alias="my_alias"),
+        ),
+        id="same alias",
+    ),
+    pytest.param(
+        build_query(
+            selected_columns=[
+                Column("_snuba_project_id", None, "project_id"),
+                Column(None, None, "transaction_name"),
+                uniq_expression(alias="different_alias"),
+            ],
+            condition=None,
+            having=uniq_expression(alias="my_alias"),
+        ),
+        id="same aggregation, different alias",
+    ),
+]
+
+
+@pytest.mark.parametrize("input_query", deepcopy(INVALID_QUERY_CASES))
```

Comment:
Why do you do a `deepcopy` here?

Reply:
It's a habit from the Query tests. Since it's just a python list it can be modified but I guess I don't need to do it here because Expressions are immutable.

Reply:
OK I was just curious if there was a reason. The list isn't modified anywhere either.